tokio/src/runtime/basic_scheduler.rs

```diff
@@ -197,6 +199,8 @@ impl SchedulerPriv {
                     self.queues.push_local(task);
                 }
             }
+
+            preemption::yielded();
```

Comment:
This probably could be moved in `task::harness` to avoid duplication in `basic_scheduler` and `thread_pool`?